### ISSUE TITLE
[feature] bookmark integration

### ISSUE THREAD

**Issue by groks** [skeeto/elfeed#110]
Emacs has a bookmark feature: C-x r m to add a bookmark to the current file, C-x r b to jump to the file where a bookmark points.

You can define new types of bookmarks for things other than files, so here's one for \*elfeed-search\*:

    (defun my-elfeed-bookmark-make-record-search ()
      "Return a bookmark record for the current elfeed-search buffer."
      `(,(format "elfeed %s" elfeed-search-filter)
        (location . ,elfeed-search-filter)
        (tags ,@(mapcar #'symbol-name
                               (nth 1 (elfeed-search-parse-filter elfeed-search-filter))))
        (handler . my-elfeed-bookmark-handler-search)))

    (defun my-elfeed-bookmark-handler-search (bmk-record)
      "Jump to an elfeed-search bookmarked location."
      (switch-to-buffer (elfeed-search-buffer))
      (unless (eq major-mode 'elfeed-search-mode)
        (elfeed-search-mode))
      (elfeed-search-set-filter (bookmark-prop-get bmk-record 'location)))

    (defun my-elfeed-search-mode-hook ()
      (setq bookmark-make-record-function #'my-elfeed-bookmark-make-record-search))

    (add-hook 'elfeed-search-mode-hook #'my-elfeed-search-mode-hook)

Emacs bookmarks do not support tags by default, but bookmark+ (in melpa) does, and the code above pre-populates the tags field.

Bookmarks could be a quick way to support more than the default elfeed-search-filter without having to create a whole new front-end.

**Comment by skeeto** [comment]
Not a bad idea. It would certainly simplify the process of creating easy-access "views" on the listing buffer. However, I don't want to create a dependency on bookmark+ just for this feature. If you can get it to work with just Emacs' built-in bookmark.el, I'd like to integrate this (and without the add-hook, since it would be part of the elfeed-search-mode itself).

**Comment by groks** [comment]
There's no dependency on bookmark+. The Emacs bookmark system allows storing extra meta-data with each bookmark. Rather than use the key 'elfeed-tags' I used 'tags' because bookmark+ takes advantage of it. Without bookmark+ the tags meta-data is ignored.

**Comment by skeeto** [comment]
Ah, gotcha! I tried your code and it didn't work, so I assumed it was
from missing bookmark+. It was only because I forgot to restart
elfeed-search-mode in my current session. This works great! It's been
committed.

Great feature, thanks!

**Comment by baudilio** [comment]
If I'm not mistaken, I think I found a bug in the bookmark/bookmark+/elfeed integration. 
I followed [skeeto's](https://github.com/skeeto/elfeed#bookmarks) procedure. When I try to create a new bookmark, I get the following error:


 Debugger entered--Lisp error: (wrong-type-argument sequencep 604800)
  mapcar(symbol-name 604800)
  elfeed-search-bookmark-make-record()
  bookmark-make-record()
  bookmark-set-internal("Set bookmark unconditionally" nil overwrite)
  bookmark-set(nil nil)
  funcall-interactively(bookmark-set nil nil)
  call-interactively(bookmark-set nil nil)
  command-execute(bookmark-set)


I suppose you'll need more information and details on the issue (OS, packages installed, emacs version, etc., etc. Before sending irrelevant data, I'll await for your directions.
